### PR TITLE
Adds PR template for iteration

### PR DESCRIPTION
**Pull request by matt-bernhardt**
## Developer

This adds a PR template for this repo.

### Tickets affected

- https://mitlibraries.atlassian.net/browse/engx-226

### Version (see config/theme.ini)

- [ ] The theme's version number has been incremented, setting up a production
      deploy.

### Documentation

- [ ] Project documentation has been updated.
- [x] No documentation changes are needed.

### Accessibility

- [ ] ANDI or WAVE has been run in accordance to [our guide](https://mitlibraries.github.io/guides/basics/a11y.html).
- [ ] New flagged issues not already resolved in this PR have been ticketed 
      (link in the Pull Request details above).
- [x] This PR contains no changes to the view layer.

### Stakeholder approval

- [ ] Stakeholder approval has been confirmed (see ticket).
- [ ] Stakeholder approval will happen retroactively.
- [x] Stakeholder approval is not needed.

### Dependencies

NO dependencies are updated

### Additional context needed to review

The linked Jira ticket describes our recent efforts to improve our PR templates.

##### Code

- [ ] I have confirmed that the code works as intended.
- [ ] Any CodeClimate issues have been fixed or confirmed as
added technical debt.

##### Documentation

- [ ] The commit message is clear and follows our guidelines
      (not just this pull request message).
- [ ] The documentation has been updated or is unnecessary.
- [ ] New dependencies are appropriate or there were no changes.

### CODE REVIEW

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+
+### Stakeholder approval
+
+- [ ] Stakeholder approval has been confirmed (see ticket).
+- [ ] Stakeholder approval will happen retroactively.
+- [ ] Stakeholder approval is not needed.
+
+### Dependencies
+
+YES | NO dependencies are updated
```
Comment (JPrevost) [inline .github/PULL_REQUEST_TEMPLATE.md:36]: What about making this more like the other check box options:

```
- [ ] Dependencies have been updated
- [ ] New dependencies have been added
- [ ] No dependencies have been added or updated

```

File: .github/PULL_REQUEST_TEMPLATE.md [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+
+
+## Code Reviewer
+
+##### Code
+
+- [ ] I have confirmed that the code works as intended.
+- [ ] Any CodeClimate issues have been fixed or confirmed as
+added technical debt.
+
```
Comment (JPrevost) [inline .github/PULL_REQUEST_TEMPLATE.md:51]: Consider added:

```
- [ ] I have reviewed any new dependencies and agree they are appropriate
```

Comment (JPrevost) [reply .github/PULL_REQUEST_TEMPLATE.md:51]: Oh. Never mind probably. I think that's covered in the next section I hadn't gotten to. I kinda like that better in the code section than docs though but either is fine.

**Review by cabutlermit — APPROVED** [review]
Looks good to me!

**Comment by matt-bernhardt** [comment]
I like your suggestions, Jeremy, so I've updated the template to reflect them. Thanks!

**Pull request merged**